webapp/channels/src/components/mfa/confirm.tsx

```diff
@@ -30,57 +30,54 @@ type Props = {
     updateParent: (state: MFAControllerState) => void;
 }
 
-export default class Confirm extends React.PureComponent<Props> {
-    public componentDidMount(): void {
-        document.body.addEventListener('keydown', this.onKeyPress);
-    }
-
-    public componentWillUnmount(): void {
-        document.body.removeEventListener('keydown', this.onKeyPress);
-    }
-
-    submit = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
+export default function Confirm({state, updateParent}: Props) {
+    const submit = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
         e.preventDefault();
         redirectUserToDefaultTeam();
     };
 
-    onKeyPress = (e: KeyboardEvent | React.FormEvent<HTMLFormElement>): void => {
-        if (isKeyPressed(e as KeyboardEvent, KeyCodes.ENTER)) {
-            this.submit(e);
+    const onKeyPress = useCallback((e: KeyboardEvent) => {
+        if (isKeyPressed(e, KeyCodes.ENTER)) {
+            submit(e);
         }
-    };
+    }, [submit]);
+
+    useEffect(() => {
+        document.body.addEventListener('keydown', onKeyPress);
+
+        return () => {
+            document.body.removeEventListener('keydown', onKeyPress);
+        };
+    }, [onKeyPress]);
 
-    public render(): JSX.Element {
-        return (
-            <div>
-                <form
-                    onSubmit={this.submit}
-                    onKeyPress={this.onKeyPress}
```

Comment:
Was it intended to remove this from here? Do we expect any unexpected result from removing it?

Reply:
I took another look at it and decided to return it so that the keyboard event can be captured on the form.